torch/csrc/inductor/aot_runtime/model.h

```diff
@@ -323,5 +324,33 @@ class AOTInductorModel : public AOTInductorModelBase<AOTInductorModel> {
   }
 };
 
+#define AOTI_TORCH_ERROR_CHECK(call)                                      \
+  if ((call) != AOTI_TORCH_SUCCESS) {                                     \
+    throw std::runtime_error(                                             \
+        std::string(#call " API call failed at ") + __FILE__ + ", line" + \
+        std::to_string(__LINE__));                                        \
+  }
+
+// For fallback ops like reinterpret_tensor which returns a short-lived
+// tensor, this RAII class will achieve a similarly short live range for
+// AtenTensorHandle in the ABI compatible mode.
+
+using RAIIAtenTensorHandle = std::shared_ptr<AtenTensorOpaque>;
+
+inline RAIIAtenTensorHandle create_raii_tensor_handle_for_extern(
+    AtenTensorHandle handle) {
+  return RAIIAtenTensorHandle(handle, [](AtenTensorHandle ptr) {
+    // Do nothing for extern tensor handles
+  });
+}
```

Comment:
It makes sense that you wanted to use this for arguments, but in fact the API for calling AOTInductor should *steal* the tensors passed into it. This ensures that tensors can be promptly deallocated as soon as AOTInductor is done using them, if there are no other references to them. If AOTInductor borrows input arguments, then they cannot deallocate until the very end. BTW, eager does it this way, that's what the .clear() on args list is for.

Reply:
Bin tells me that upstream requested a borrow-only interface. That seems a bit strange to me. @zdevito do you know if TorchScript supports operators that steal their inputs?

Reply:
Even if one integration wants to only send borrows to AOTInductor, I still think the model.so API should steal in tensors. If you don't do this, then it is impossible to support stealing without a stable API breakage. But if you take steals on model.so, the top level can always bump the refcount by one before sending the tensor in.